### DISCUSSION TITLE
SSH authentication between server and clients

### DISCUSSION THREAD

**Discussion by ikki0123** [ltsp/ltsp#731] [General] [answered]
Hello, 
In my project I need authentication via ssh between clients and servers but I'm having problems.
I usually generate the public keys and put them in the athorized_keys file to authenticate without passwords.
My first problem was that in the initialization of the clients they lost their keys, having to generate them by the command ssh-keygen -A. I managed to get around this problem by putting some lines in ltsp.conf , KEEP_SYSTEM_SERVICES="ssh"
POST_INIT_CONFIG_SSH="dpkg-reconfigure openssh-server", this solved my problem regarding SSH service startup on clients.

But now I don't know what to do to solve the SSH authentication problem between clients and server. Every time I restart the clients are generated new keys.
Is there anything that can be done about this?

Sorry for the bad english.

**Comment by alkisg** [comment, marked as answer]
First of all, exposing the private SSH keys into a world-readable squashfs x86_64.img image is a security issue. That's why LTSP recommends using https://epoptes.org instead, which uses reverse connections to the server, and requires no client-side secrets.

That said, if you really want to enable sshd on LTSP clients, see the [previous discussion about it](https://github.com/ltsp/ltsp/discussions/310#discussioncomment-101549).

**Reply by ikki0123** [reply]
Thank you again! It was just what I was needing. 
I ended up using SSH even followed the link you mentioned
